### PR TITLE
Remove writes on validation failure

### PR DESCRIPTION
**Pull request by mbj**
* It does not make sense to attempt to write to an invalid record on
  detecting that invalid state.
* Back in the history the writes on #reset_checkout_flow where
  done bypasisng the model via  #update_columns to reset the order
  state. This was even more ideotic as all totals where not reset etc.
* Its better to simply tell the customer: No I do not checkout the cart
  for you because $reason, and not reset the checkout state. The user
  has to nagivate back to the cart manually *anyway* to fix it,
  regardless of the Order#state value, as writing it does not force
  navigation at all.

Extraction from #398

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: core/app/models/spree/order.rb [hunk 1]
```diff
@@ -344,7 +344,6 @@ def insufficient_stock_lines
     def ensure_line_item_variants_are_not_deleted
       if line_items.any?{ |li| !li.variant || li.variant.paranoia_destroyed? }
         errors.add(:base, Spree.t(:deleted_variants_present))
-        restart_checkout_flow
```
Comment (dkubb) [inline core/app/models/spree/order.rb]: Not a blocker, and I know you didn't want to touch any code unnecessarily, but I wonder if this method could be rewritten using a guard clause, eg:

```ruby
def ensure_line_item_variants_are_not_deleted
  return true unless line_items.any?(&:paranoia_destroyed?)
  errors.add(:base, Spree.t(:deleted_variants_present))
  false
end
```

Also given my fix in #485 and the true foreign key the line item variant will always exist, so we don't have to test for a nil variant.

File: core/app/models/spree/order.rb [hunk 2]
```diff
@@ -354,7 +353,6 @@ def ensure_line_item_variants_are_not_deleted
     def ensure_line_items_are_in_stock
       if insufficient_stock_lines.present?
         errors.add(:base, Spree.t(:insufficient_stock_lines_present))
-        restart_checkout_flow
```
Comment (dkubb) [inline core/app/models/spree/order.rb]: Instance of https://github.com/MountainRoseHerbs/spree/pull/495/files#r77856950

File: core/app/models/spree/order.rb (hunk 1 above)
Comment (mbj) [inline core/app/models/spree/order.rb]: @dkubb I agree with that change, but lets do it in a dedicated iteration.

**Pull request merged**